docs/user-guide/CAI.rst

```diff
@@ -0,0 +1,308 @@
+      model.micro_batch_size=1 \
+      model.global_batch_size=512 \
+      ++model.activations_checkpoint_granularity="full" \
+      ++model.activations_checkpoint_method="uniform" \
+      ++model.activations_checkpoint_num_layers=1 \
+      model.optim.bucket_cap_mb=200 \
+      model.data.seq_length=4096 \
+      model.data.data_impl=json \
+      exp_manager.explicit_log_dir=<path to output dir> \
+      exp_manager.checkpoint_callback_params.save_top_k=1 \
+      exp_manager.checkpoint_callback_params.save_nemo_on_train_end=True \
+      trainer.rm.save_interval=25 \
+      trainer.rm.val_check_interval=25 \
+      trainer.rm.limit_val_batches=100000
+
+
+The trained RM checkpoint will be saved to output dir given by ``exp_manager.explicit_log_dir``.
+
+Step 6: Finetune Mistral-7B-SL-CAI with PPO and the RM to train a Mistral-7B-RL-CAI model
+##############################################################################################
+Run the following command in the background to launch a RM and PPO critic training server:
+
+.. code-block:: bash
+
+   python examples/nlp/gpt/serve_ppo_critic.py \
+      pretrained_checkpoint.restore_from_path=<path to pretrained model, e.g., mistral_7b_sl_cai_rm.nemo> \
+      trainer.ppo.inference_micro_batch_size=4 \
+      trainer.ppo.port=5567 \
+      model.offload_adam_states=True \
+      model.micro_batch_size=1 \
+      model.global_batch_size=64 \
+      model.seed=1234 \
+      exp_manager.explicit_log_dir=<path to critic output dir>
+
+Run the following command to launch actor training and reference policy server:
+
+.. code-block:: bash
+
+   python -u examples/nlp/gpt/train_gpt_ppo_actor.py \
+      pretrained_checkpoint.restore_from_path=/path/to/sft_log_dir/checkpoints/megatron_gpt_sft.nemo \
+      model.data.data_impl=json \
+      "model.data.data_prefix={train: [<path to cai_preference_dataset_prompts_train.jsonl>], validation: [<path to cai_preference_dataset_prompts_test.jsonl>], test: [<path to cai_preference_dataset_prompts_test.jsonl>]}" \
+      exp_manager.explicit_log_dir=<path to actor output dir> \
+      model.micro_batch_size=1 \
+      model.global_batch_size=64 \
+      ++model.activations_checkpoint_granularity=selective \
+      ++model.activations_checkpoint_method=uniform \
+      model.optim.lr=9e-8 \
+      model.optim.sched.min_lr=9e-9 \
+      model.ppo.num_rollout_samples=512 \
+      model.ppo.rollout_micro_batch_size=8 \
+      model.ppo.length_params.max_length=1024 \
+      trainer.ppo.initial_policy_kl_penalty=0.02 \
+      remote_critic_rm.critic.ip=<ip to critic service> \
+      remote_critic_rm.critic.port=5567
+
+The trained policy (LLM) checkpoint will be saved to output dir given by ``exp_manager.explicit_log_dir``.
```

Comment:
The trained LLM policy checkpoint will be saved to the output dir given by . . .